packages/@aws-cdk/toolkit-lib/lib/toolkit/toolkit.ts

```diff
@@ -375,6 +378,85 @@ export class Toolkit extends CloudAssemblySourceBuilder {
     return templateDiffs;
   }
 
+  /**
+   * Drift Action
+   */
+  public async drift(cx: ICloudAssemblySource, options: DriftOptions): Promise<DriftCommandResult> {
+    const ioHelper = asIoHelper(this.ioHost, 'drift');
+    const selectStacks = options.stacks ?? ALL_STACKS;
+    await using assembly = await assemblyFromSource(ioHelper, cx);
+    const stacks = await assembly.selectStacksV2(selectStacks);
+
+    let drifts = 0;
+    let uncheckedResources = 0;
+    const output = [];
+
+    for (const stack of stacks.stackArtifacts) {
+      const cfn = (await (await this.sdkProvider('drift')).forEnvironment(stack.environment, Mode.ForReading)).sdk.cloudFormation();
+
+      const driftResults = await detectStackDrift(
+        cfn,
+        ioHelper,
+        stack.stackName,
+      );
+
+      // Get all stack resources if we want to show all
+      const allStackResources = new Map<string, string>();
+      Object.keys(stack.template.Resources || {}).forEach(id => {
+        const resource = stack.template.Resources[id];
+        allStackResources.set(id, resource.Type);
+      });
+
+      if (!driftResults) {
+        await ioHelper.notify(IO.CDK_TOOLKIT_I4505.msg(`${chalk.bold(stack.displayName)}: unable to detect drift`));
+        continue;
+      }
+
+      const formatter = new DriftFormatter({
+        ioHelper: asIoHelper(this.ioHost, 'drift'),
```

Comment:
You already have one
```suggestion
        ioHelper,
```